### PR TITLE
Add test for POST /api/card/:id/query for parameters with defaults

### PR DESCRIPTION
**Pull request by calherries**
This PR adds a test to clarify the BE behaviour of `POST /api/card/:id/query` when there are defaults applied to parameters. This behaviour was implemented in https://github.com/metabase/metabase/pull/31891, but there were no tests specifically for this endpoint.

The test is mostly redundant because the same code path is used for other endpoints e.g. `POST /api/dashboard/:dashboard-id/card/:card-id/query`, which are tested and the code that does default parameter substitution is tested [here](https://github.com/metabase/metabase/pull/31891/files#diff-ce58525a897b47b32a26ef9067c537a410de2e1e006ef9f6b1ec5443d1dd5bc1). But since it's pretty easy for the FE to use this endpoint incorrectly (see the confusion caused by https://github.com/metabase/metabase/issues/37831), I thought it would be nice to add a test for this endpoint to document and lock down it's behaviour.

### CODE REVIEW

File: test/metabase/api/card_test.clj [hunk 1]
```diff
@@ -189,6 +189,46 @@
...
+                                                                 :id           "_VENUE_ID_",
+                                                                 :name         "venue_id",
+                                                                 :required     false,
+                                                                 :default      1
+                                                                 :type         :dimension,
+                                                                 :widget-type  :id}}}}}]
+      (let [request (fn [body]
+                      (mt/user-http-request :rasta :post 202 (format "card/%d/query" (:id card)) body))]
+        (testing "the default can be overridden"
+          (is (= [[1]]
```
Comment (adam-james-v) [inline test/metabase/api/card_test.clj:210]: Should this be `[[2]]` or am I misunderstanding the test?

File: test/metabase/api/card_test.clj [hunk 2]
```diff
@@ -189,6 +189,46 @@
...
+                 (mt/rows (request {:parameters [{:id    "_VENUE_ID_"
+                                                  :name  "venue_id"
+                                                  :slug  "venue_id"
+                                                  :type  "number"
+                                                  :value 2}]})))))
+        (testing "the default should apply if no param value is provided"
+          (is (= [[1]]
+                 (mt/rows (request {:parameters []}))))
```
Comment (oleggromov) [inline test/metabase/api/card_test.clj:218]: What if parameter is passed but its value is unset?

Comment (calherries) [reply test/metabase/api/card_test.clj:218]: `value: null` and omitting the `value` key are equivalent.

Comment (oleggromov) [reply test/metabase/api/card_test.clj:218]: Got it, thanks.
Don't we have a convention somewhere that passing `nil` as a value (`null` in json) means "we deliberately want the value to be empty" while making it unset (`undefined` or no property in json) means "take default"?

File: test/metabase/api/card_test.clj (hunk 1 above)
Comment (calherries) [reply test/metabase/api/card_test.clj]: Oleg's right but it wasn't a strong test at all. Thanks for pointing this out, I've updated it now.

File: test/metabase/api/card_test.clj (hunk 2 above)
Comment (calherries) [reply test/metabase/api/card_test.clj:217]: I'm pretty sure that was the behaviour before https://github.com/metabase/metabase/pull/31891, but now it's like this. Having a semantic difference between `value: null` and omitting the `value` key is asking for trouble, especially in Clojure where most functions will treat them the same.

**Comment by calherries** [comment]
@adam-james-v can you take another look?

**Review by adam-james-v — APPROVED** [review]
Looks good :)

**Pull request merged**